server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/utils/quotas/QuotaLoaderWithDefaultPreloaded.scala

```diff
@@ -0,0 +1,61 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils.quotas
+
+import org.apache.james.jmap.mail.{Quota, QuotaId, QuotaRoot, Quotas}
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.mailbox.exception.MailboxException
+import org.apache.james.mailbox.model.{MailboxPath, QuotaRoot => JavaQuotaRoot}
+import org.apache.james.mailbox.quota.{QuotaManager, UserQuotaRootResolver}
+
+class QuotaLoaderWithDefaultPreloaded (quotaRootResolver: UserQuotaRootResolver,
+                                       quotaManager: QuotaManager,
+                                       session: MailboxSession) extends QuotaLoader {
+  @throws[MailboxException]
+  private val preloadedUserDefaultQuotas: Quotas = getUserDefaultQuotas
+
+  @throws[MailboxException]
+  override def getQuotas(mailboxPath: MailboxPath): Quotas = {
+    val quotaRoot: JavaQuotaRoot = quotaRootResolver.getQuotaRoot(mailboxPath)
+    val quotaId: QuotaId = QuotaId.fromQuotaRoot(QuotaRoot.fromJava(quotaRoot))
+
+    if (mailboxPath.getUser.equals(session.getUser)) {
+      return preloadedUserDefaultQuotas
+    }
+
+    Quotas.from(
+      quotaId,
+      Quota.from(Map(
+        Quotas.Storage -> quotaToValue(quotaManager.getStorageQuota(quotaRoot)),
+        Quotas.Message -> quotaToValue(quotaManager.getMessageQuota(quotaRoot)))))
+  }
+
+  @throws[MailboxException]
+  private def getUserDefaultQuotas: Quotas = {
+    val quotaRoot: JavaQuotaRoot = quotaRootResolver.forUser(session.getUser)
+    val quotaId: QuotaId = QuotaId.fromQuotaRoot(QuotaRoot.fromJava(quotaRoot))
+
+    Quotas.from(
+      quotaId,
+      Quota.from(Map(
+        Quotas.Storage -> quotaToValue(quotaManager.getStorageQuota(quotaRoot)),
+        Quotas.Message -> quotaToValue(quotaManager.getMessageQuota(quotaRoot)))))
+  }
```

Comment:
Duplicated with the above code.

We will benefit from a `def retrieveQuotas(quotaRoot: quotaRoot): Quotas` method